### ISSUE TITLE
`Partition.make`: reconsider `ty` argument

### ISSUE THREAD

**Issue by reynir** [mirage/ocaml-mbr#12]
At the moment `Partition.make` defaults to partition type 0x06 (FAT16). This default made sense years ago. Today I think it's not a good default. I think making the type a required argument is more sensible.

**Comment by AryanGodara** [comment]
Hello @reynir, I would love to work on this issue.
Can you please assign it to me! (I'm from Outreachy, and introduced myself on discord channel)

**Comment by reynir** [comment]
Yes! I assigned you to this issue. You may want to skim the [Wikipedia article](https://en.wikipedia.org/wiki/Master_boot_record#PT). Only "modern standard MBR" is relevant.

If you are motivated and have time it would be great if you could look at #1 as well as it is related and not a big change.

**Comment by AryanGodara** [comment]
Thank you @reynir , I'll start working on this right away!
Yes, I'll go through #1 as well!

**Comment by AryanGodara** [comment]
This is taking little longer to get started, as I'm not very comfortable with OCaml, and have midsems going on.
I'm still working on this issue, just need some time!
Will open PR soon, or ask for help when stuck!

**Comment by reynir** [comment]
Of course! Take your time. This book on OCaml is available for free and is a good book to start with https://ocaml-book.com/

**Comment by AryanGodara** [comment]
The book is  really helpful @reynir , after going through the documentation before this book, I'm finally getting the hang of functional programming😅 and oCaml and the tools

**Comment by reynir** [comment]
The function definition starts here:

https://github.com/mirage/ocaml-mbr/blob/master/lib/mbr.ml#L73

The module `Partition` is defined here:

https://github.com/mirage/ocaml-mbr/blob/master/lib/mbr.ml#L58-L153

To refer to functions defined inside `Partition` you have to add `Partition.` in front, for example `Partition.make`.

**Comment by AryanGodara** [comment]
> The function definition starts here:
> 
> https://github.com/mirage/ocaml-mbr/blob/master/lib/mbr.ml#L73
> 
> The module `Partition` is defined here:
> 
> https://github.com/mirage/ocaml-mbr/blob/master/lib/mbr.ml#L58-L153
> 
> To refer to functions defined inside `Partition` you have to add `Partition.` in front, for example `Partition.make`.

Thanks @reynir , I found the function, and am now updating it.
